scrapinghub/client.py

```diff
@@ -0,0 +1,350 @@
+
+
+class Projects(object):
+
+    def __init__(self, client):
+        self._client = client
+
+    def get(self, projectid):
+        return Project(self._client, parse_project_id(projectid))
+
+    def list(self):
+        return self._client._connection.project_ids()
+
+    def summary(self, **params):
+        return self._client._hsclient.projects.jobsummaries(**params)
+
+
+class Project(object):
+
+    def __init__(self, client, projectid):
+        self.id = projectid
+        self._client = client
+
+        # sub-resources
+        self.jobs = Jobs(client, projectid)
+        self.spiders = Spiders(client, projectid)
+
+        # proxied sub-resources
+        self.activity = Activity(client._hsclient, projectid)
+        self.collections = Collections(_Collections, client, projectid)
+        self.frontier = Frontier(client._hsclient, projectid)
+        self.reports = Reports(client._hsclient, projectid)
+        self.settings = Settings(client._hsclient, projectid)
+
+
+class Spiders(object):
+
+    def __init__(self, client, projectid):
+        self.projectid = projectid
+        self._client = client
+
+    def get(self, spidername, **params):
+        project = self._client._hsclient.get_project(self.projectid)
+        spiderid = project.ids.spider(spidername, **params)
+        if spiderid is None:
+            raise NonExistingSpider("Spider {} doesn't exist."
+                                    .format(spidername))
+        return Spider(self._client, self.projectid, spiderid, spidername)
+
+    def list(self):
+        project = self._client._connection[self.projectid]
+        return project.spiders()
+
+
+class Spider(object):
+
+    def __init__(self, client, projectid, spiderid, spidername):
+        self.projectid = projectid
+        self.id = spiderid
+        self.name = spidername
+        self.jobs = Jobs(client, projectid, self)
+        self._client = client
+
+    def update_tags(self, add=None, remove=None):
+        params = get_tags_for_update(add_tag=add, remove_tag=remove)
+        if not params:
+            return
+        params.update({'project': self.projectid, 'spider': self.name})
+        result = self._client._connection._post('jobs_update', 'json', params)
+        return result['count']
+
+
+class Jobs(object):
+
+    def __init__(self, client, projectid, spider=None):
+        self.projectid = projectid
+        self.spider = spider
+        self._client = client
+
+    @property
+    def _hsproject(self):
+        """Shortcut to hsclient.project for internal use"""
+        return self._client._hsclient.get_project(self.projectid)
+
+    def count(self, **params):
+        if self.spider:
+            params['spider'] = self.spider.name
+        return next(self._hsproject.jobq.apiget(('count',), params=params))
+
+    def iter(self, **params):
+        """ Iterate over jobs collection for a given set of params.
+        FIXME the function returns a list of dicts, not a list of Job's
```

Comment:
Agreed that `iter` returns jobs summary, not jobs.
Let's return to the question when there's any real need for it